packages/imperative/src/plugins/PluginManagementFacility.ts

```diff
@@ -25,6 +25,7 @@ import { ConfigurationValidator } from "../ConfigurationValidator";
 import { ConfigurationLoader } from "../ConfigurationLoader";
 import { DefinitionTreeResolver } from "../DefinitionTreeResolver";
 import { IImperativeOverrides } from "../doc/IImperativeOverrides";
+import { ICredentialManager } from "../../../settings/src/doc/ISettingsFile";
```

Comment:
Tested changes to PluginManagementFacility with the K8s secrets plug-in.

With the following config, the "plugin" property is ignored and the CLI still stores credentials with Keytar:
```json
{
  "overrides": {
    "CredentialManager": {
        "plugin": "@zowe/secrets-for-kubernetes-for-zowe-cli",
        "type": "keytar"
    }
  }
}
```

With the following config, I get an error `Failed to load the credential manager named "[object Object]"`:
```json
{
  "overrides": {
    "CredentialManager": {
        "plugin": "@zowe/secrets-for-kubernetes-for-zowe-cli",
        "type": "k8s"
    }
  }
}
```

Is this the intended behavior?

Reply:
Tested this case and it seems to be using k8s just fine, it seems like the PluginManagementFacility will view that a plugin is available or installed and will just pass in the class required to load instead of the override regardless of the type `keytar` passed in if loading with the name of the installed plugin

Reply:
I'm afraid I haven't tested this yet, but I can see why that could happen (if there is an error in the plugin configuration)
![image](https://user-images.githubusercontent.com/37381190/218568106-329e83e9-f6e5-4d34-94c0-9dbd07291cd6.png)